### PR TITLE
to_timestamp i32 coerced to i64

### LINKED ISSUE #3046: incorrect i32 coercion for `to_timestamp`
**Describe the bug**

it's coerced to utf-8

**To Reproduce**

```bash
❯ select to_timestamp(cast(1.0 as int));
ArrowError(CastError("Error parsing '1' as timestamp"))
```

**Expected behavior**

coerced to int64

**Additional context**

### PR DESCRIPTION
**Pull request by waitingkuo**
# Which issue does this PR close?

Closes #3046 

 # Rationale for this change

# What changes are included in this PR?

change the oreder for `to_timestamp` signagure. make int64 be in front of utf8

# Are there any user-facing changes?

### CODE REVIEW

**Review by waitingkuo — COMMENTED** [review]
this is the fix

**Comment by waitingkuo** [comment]
could anyone help me to figure this error? (for this first check `Dev / Lint C++, .......`)

```bash
Run archery lint --rat
  archery lint --rat
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/apache/arrow-datafusion/runs/7698281962?check_suite_focus=true#step:5:3).10.5/x6[4](https://github.com/apache/arrow-datafusion/runs/7698281962?check_suite_focus=true#step:5:4)
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.[5](https://github.com/apache/arrow-datafusion/runs/7698281962?check_suite_focus=true#step:5:5)/x[6](https://github.com/apache/arrow-datafusion/runs/7698281962?check_suite_focus=true#step:5:6)4/lib
INFO:archery:Running apache-rat linter
apache-rat license violation: js/test.ts
Error: Process completed with exit code 1.
```

**Comment by waitingkuo** [comment]
@alamb  would you mind helping me figure out my first check? I've committed dummy commits to trigger this several times.

**Comment by alamb** [comment]
https://github.com/apache/arrow-datafusion/runs/7699097645?check_suite_focus=true


> Run archery lint --rat
> INFO:archery:Running apache-rat linter
> apache-rat license violation: js/test.ts

This means (very unobviously) that the file `js/test.ts` has no license header

Interestingly, this test is also failing on master changes too -- I will get a PR up to fix it
https://github.com/apache/arrow-datafusion/runs/7698061920?check_suite_focus=true

**Comment by alamb** [comment]
https://github.com/apache/arrow-datafusion/issues/3052 tracks the CI failure

**Comment by alamb** [comment]
https://github.com/apache/arrow-datafusion/pull/3053 should "fix" the problem -- TLDR is that the the CI check failure is not related to your changes in this PR @waitingkuo

File: datafusion/core/tests/sql/timestamp.rs [hunk 1]
```diff
@@ -1065,3 +1065,23 @@ async fn cast_to_timestamp_micros_twice() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn to_timestamp_i32() -> Result<()> {
+    let ctx = SessionContext::new();
+
+    let sql = "select to_timestamp(cast (1 as int));";
```
Comment (alamb) [inline datafusion/core/tests/sql/timestamp.rs:1073]: Can we please add `to_timestamp_seconds(..)`, `to_timestamp_milliseconds(..)` etc here?

**Review by alamb — COMMENTED** [review]
Thanks @waitingkuo  -- other than the test coverage looks good to me!

**Review by alamb — APPROVED** [review]
LGTM -- thanks @waitingkuo

**Pull request merged**